### ISSUE TITLE
Betterdiscord... Icon?

### ISSUE THREAD

**Issue by d34ddud3** [bb010g/betterdiscordctl#78]
So what does this flashing Better Discord icon in the bottom right mean?
![image](https://user-images.githubusercontent.com/70095654/91118003-971b9f80-e655-11ea-8f62-d26ced25f61a.png)

Notes: 
I can't click on it.
Betterdiscord shows no other signs of working. (I've made use of themes and plugins in the past)
I'm using Debian 10 and have followed all instructions given on installing it.
I have gotten it to work previously, but it seems to randomly work and then not work between reboots.

**Comment by ObserverOfTime** [comment]
Run `betterdiscordctl reinstall` and `betterdiscordctl update`.
If it still fails to load, it's probably a network issue.

**Comment by d34ddud3** [comment]
I did just do it, but man the number of times I've reinstalled both Discord and Betterdiscord. It still doesn't work, but the Icon is still there. I did make another post yesterday (I think it was deleted or something because I couldn't find it) and there are two things that I think could be the problem. One is that I was trying to install an emulator, had issues, and went to my BIOS to enable it. The other is a line I get when told it cannot find Discord and it recommends using "--discord". Then I used it and it said something about a config file. My other post was much more detailed, like I said, but I couldn't find it. I can try looking for it again to see if I can get that line.

**Comment by ObserverOfTime** [comment]
So…where's that detailed post?

**Comment by d34ddud3** [comment]
I couldn't find it unfortunately. It should have been posted here. But as a summary I used [this link](https://linuxconfig.org/how-to-install-discord-on-linux) to install Discord. It says it should be Canary but I tried to install Betterdiscord for Canary and it didn't work (It says "Discord installation not found."). So I tried the normal version and it worked _kind of._ I had it working properly a couple times and it would randomly work or break between reboots. Although it doesn't do that anymore, it doesn't work at all now. Though I had uninstalled, reinstalled, updated, and upgraded repeatedly when it was working even that little bit. It also happened to stop working at all when I enabled use of virtual machines in the BIOS. And like I said previously, there's also some error it gives about the config file. Though it's been too long, I can't remember what exactly it said. What I have noticed is the following regarding Betterdiscord showing signs of working: The flashing icon from above and when I hit "public" under the friend icon up at the top left I get the Betterdiscord server option, but I can't join it. Though I can join other ones.
![image](https://user-images.githubusercontent.com/70095654/91668115-79c05880-eacf-11ea-9411-74602b4c678d.png)

**Comment by ObserverOfTime** [comment]
> there's also some error it gives about the config file
Would it happen to be the same error as #77?

**Comment by d34ddud3** [comment]
I just took a look and no, Discord functions perfectly fine. I don't have a problem opening that. I have also only used sudo when the instructions said to, and I've been paying attention to whether the lines start with a # or a $. (I never really use super user I just use sudo.) I _might_ have used it the first time I installed Betterdiscord, but I've been typing betterdiscordctl commands without sudo as far as I remember. I tried as you instructed anyway. I deleted `.local/share/betterdiscordctl` and tried again.

`user@hostname:~$ betterdiscordctl install`
`Cloning https://github.com/rauenzi/BetterDiscordApp...`
`Cloning into '/home/d34ddud3/.local/share/betterdiscordctl/bd/0'...`
`remote: Enumerating objects: 13, done.`
`remote: Counting objects: 100% (13/13), done.`
`remote: Compressing objects: 100% (10/10), done.`
`remote: Total 13 (delta 0), reused 6 (delta 0), pack-reused 0`
`Unpacking objects: 100% (13/13), done.`
`Installed. (Restart Discord if necessary.)`
`user@hostname:~$ betterdiscordctl upgrade`
`betterdiscordctl is already the latest version (1.7.0).`
`user@hostname:~$ betterdiscordctl update`
`Updating Git repository...`
`From https://github.com/rauenzi/BetterDiscordApp`
` * branch            injector   -> FETCH_HEAD`
`HEAD is now at 79ff7d8 change backup url`
`user@hostname:~$`

It seems like it works from here, but when I restart Discord the flashing icon and "public" button under the friends icon to access dm's have both disappeared. Now there are absolutely no signs of it working other than what the console says above. I tried once to install Canary as well after this because the website I linked in my last comment said it would be Canary. Though Betterdiscord doesn't seem to install if I try the Canary installation. (I tried after deleting `.local/share/betterdiscordctl` again before trying just to make sure.)

**Comment by ObserverOfTime** [comment]
Run `betterdiscordctl status -v -v` and post the output.

For Canary you should add the `--flavor=canary` flag.

**Comment by d34ddud3** [comment]
`user@hostname:~$ betterdiscordctl -v -v`
`VV: Scanning /opt`
`VV: Trying flavor ''`
`VV: Trying flavor 'canary'`
`VV: Trying flavor 'ptb'`
`VV: Scanning /usr/share`
`VV: Trying flavor ''`
`V: Using Discord at /usr/share/discord`
`V: Found modules in /home/d34ddud3/.config/discord/0.0.11/modules`
`Discord: /usr/share/discord`
`Modules: /home/d34ddud3/.config/discord/0.0.11/modules`
`Index modified: yes`
`Linked injector directory: /home/d34ddud3/.local/share/betterdiscordctl/bd/0`
`Linked injector repository: https://github.com/rauenzi/BetterDiscordApp`

I tried to use the `--flavor=canary` but I don't know how. I tried adding it to `betterdiscordctl` and `betterdiscordctl -v -v` but it just spat out the line `WARN: Unknown option (ignored): --flavor=canary` and ran the rest of the command. I tried looking through `betterdiscordctl -h` and a Google search (that ended up as fruitless as a desert) but I couldn't figure out how to use it.

**Comment by ObserverOfTime** [comment]
My bad, it's <code>--flavor<b>s</b>=canary</code>.

**Comment by d34ddud3** [comment]
Alright... I am confused beyond measure now. So when I saw your comment I decided to look into whether that website lied to me or not, turns out it did. I have normal, apparently Canary has an orange icon which I didn't know because I only ever used the stable and standard. So while I was looking at Discord I noticed, _"oh hey, I have an update. Maybe that will fix it?"_ and right I was! _Almost..._ After the update I decided to reboot to see if the previous problem would still occur... _and it did..._ So now I once again have the flashing icon at the bottom right and the "public" button under the friend dm's icon at the top left... I don't think I've ever been so confused before in my life. _Sidenote: That update was apparently from August 10th... is there an option to turn on update notifications or auto-updates? If not is there a plugin I can get? And if still no that would be an awesome feature to add to Betterdiscord because that update icon is very easy to miss and forget about._

**Comment by d34ddud3** [comment]
So I just installed the Discord update file `discord-0.0.12.deb` and completely reinstalled BetterDiscord (without sudo as always) in the order of install, upgrade (1.7.0), then update. Now I see absolutely no signs of it working at all. I assume the most likely reason is that I just need to wait for the next BetterDiscord version but I figure this post wouldn't hurt.

**Comment by ObserverOfTime** [comment]
So...there's no issue?

**Comment by d34ddud3** [comment]
No, I'm just back to where I started when I made the initial post. I have no BetterDiscord settings. The only difference from when I started is that the `public` button is in the right place and I updated Discord from `discord-0.0.11.deb` to `discord-0.0.12.deb`.

**Comment by ObserverOfTime** [comment]
Open the Discord console with `Ctrl` + `Shift` + `i` and screenshot any errors you see.

**Comment by d34ddud3** [comment]
![image](https://user-images.githubusercontent.com/70095654/93028500-ebbd9500-f5d9-11ea-8730-0d8b866135a0.png)
These are the only two warnings and errors I have.

**Comment by ObserverOfTime** [comment]
Can you upload `~/.config/BetterDiscord/injector.log` and `~/.config/BetterDiscord/bdstorage.json`?

**Comment by d34ddud3** [comment]
I can confirm the existence of one of those things. Here's `bdstorage.json`. (It wouldn't let me upload file directly, so I copy and pasted through Kate)

`{`
`    "settings": {`
`        "stable": {`
`            "settings": {`
`                "bda-gs-1": true,`
`                "bda-gs-2": false,`
`                "bda-gs-3": false,`
`                "bda-gs-4": false,`
`                "bda-gs-5": true,`
`                "bda-gs-6": true,`
`                "bda-gs-7": true,`
`                "bda-gs-8": false,`
`                "bda-es-0": true,`
`                "bda-es-1": true,`
`                "bda-es-2": true,`
`                "bda-es-4": false,`
`                "bda-es-6": true,`
`                "bda-es-7": true,`
`                "bda-gs-b": false,`
`                "bda-es-8": true,`
`                "bda-dc-0": false,`
`                "bda-css-0": false,`
`                "bda-css-1": false,`
`                "bda-es-9": true,`
`                "fork-dm-1": false,`
`                "fork-ps-1": true,`
`                "fork-ps-2": true,`
`                "fork-ps-3": true,`
`                "fork-ps-4": true,`
`                "fork-ps-5": true,`
`                "fork-es-2": false,`
`                "fork-es-3": true,`
`                "fork-wp-1": false,`
`                "fork-wp-2": false,`
`                "fork-beta": true,`
`                "reactDevTools": false`
`            },`
`            "plugins": {},`
`            "themes": {`
`                "theme1": true,`
`                "theme2": false,`
`                "theme3": false,`
`                "theme4": false,`
`                "theme5": false,`
`                "theme6": false,`
`                "theme7": false`
`            }`
`        },`
`        "canary": {},`
`        "ptb": {}`
`    },`
`    "version": "0.3.5",`
`    "emoteCacheDays": 7,`
`    "emoteCacheDate": "2020-09-11T21:10:47.833Z"`
`}`

As for `injector.log`... Yeah that doesn't exist. 
![image](https://user-images.githubusercontent.com/70095654/93068679-7043fe00-f642-11ea-9e26-1b8ab35e81fa.png)

So I decided to run `betterdiscordctl update` after I saw that.
`user@hostname:~$ betterdiscordctl update`
`Updating Git repository...`
`From https://github.com/rauenzi/BetterDiscordApp`
` * branch            injector   -> FETCH_HEAD`
`HEAD is now at 79ff7d8 change backup url`
`user@hostname:~$`
The last line caught my eye. "...change backup url" Might that be the problem with `injector.log` not appearing?

**Comment by ObserverOfTime** [comment]
No. Run `betterdiscordctl install -v -v` now and post the output if it fails

**Comment by d34ddud3** [comment]
`user@hostname:~$ betterdiscordctl install -v -v`
`VV: Scanning /opt`
`VV: Trying flavor ''`
`VV: Trying flavor 'canary'`
`VV: Trying flavor 'ptb'`
`VV: Scanning /usr/share`
`VV: Trying flavor ''`
`V: Using Discord at /usr/share/discord`
`V: Found modules in /home/user/.config/discord/0.0.12/modules`
`ERROR: Already installed.`
`user@hostname:~$`

I went and checked for `injector.log` after this and nothing changed there.

**Comment by ObserverOfTime** [comment]
Again with `reinstall` instead.

**Comment by d34ddud3** [comment]
`user@hostname:~$ betterdiscordctl reinstall -v -v`
`VV: Scanning /opt`
`VV: Trying flavor ''`
`VV: Trying flavor 'canary'`
`VV: Trying flavor 'ptb'`
`VV: Scanning /usr/share`
`VV: Trying flavor ''`
`V: Using Discord at /usr/share/discord`
`V: Found modules in /home/d34ddud3/.config/discord/0.0.12/modules`
`Killing Discord processes...`
`Discord killed (pid 888)`
`Discord killed (pid 2038)`
`Discord killed (pid 2041)`
`Discord killed (pid 2043)`
`Discord killed (pid 2065)`
`Discord killed (pid 2089)`
`Discord killed (pid 2123)`
`V: Removing old injector folder.`
`V: Injecting into index.js...`
`V: Linking BetterDiscord injector...`
`Reinstalled.`
`user@hostname:~$`

So by doing this BetterDiscord seems to work now. The `public` button is misplaced again but that's a simple fix in the `.css` files I suppose. I went ahead and did a reboot to see if it broke again and it didn't. I then went into Discord and used `ctrl+shift+i` and checked the console tab. The two errors went away, but those same two warnings remain. Don't know if that's important or not. And the file `injector.log` still doesn't exist under `~/.config/BetterDiscord`. That directory hasn't changed at all. I suppose it's probably not too important in terms of function.
